### DISCUSSION TITLE
TMC Connection Error Hemera Hotend

### DISCUSSION THREAD

**Discussion by dkoke** [Jyers/Marlin#879] [Q&A] [answered]
Hello!

I believe the TMC connection error is a result of the Hemera stepper motor. I beleive it is the hotend stepper motor because I noticed I was under extruding badly. When I went to calculate the E Steps by extruding 100mm of filament it only extruded out 33mm of filament. 

I was hoping someone knew what tweaks needed to be made to the software in order for it to resolve the tmc connection error and extrusion issue. I double checked the firmware and ran M503 to ensure my E steps were 409 as perscribed by E3D.

Thank you for the help!

**Comment by msdossajo** [comment]
I'm using a BMG clone with my 4.2 "B" that uses TMC2209 drivers, after converting the board to use Linear Advance (at that time I was unaware that my board had 2209) I had to increase the esteps value to 796.
Probably you also need to increase the value. I'm also using a pancake motor with a Vref of 0.9V and it prints without skipping steps. 
I also have the TMC Connection Error after converting the board, just ignore the error.

**Reply by Reggi3** [reply]
It might be that it's something to do with the microsteps setting in configuration_adv.h ?  Is the stepper 1.8degrees per step or 0.9degrees/step?

**Comment by dkoke** [comment, marked as answer]
Okay, so instead of compiling the firmware with the Jyers config.h and config.adv.h inside the configuration folder. I instead went through and configured with the config.h and config.adv.h that is already present in the Marling folder. 

I could not narrow it down to what caused the TMC connection error, but I did resolve my issue by doing the following above. I kept the same ender 3v2 vref values and recommended steps. 

Everything extrudes normally. Im within 1-2mm on the e-step calibration test. 

Thank you all for the help!

**Reply by Bhumpe** [reply]
If you don't want to rewrite all, you need to comment line 757 in Configuration.h, that line causes the error.
#define TMC_UART_PIN_E PA13

**Reply by smg492** [reply]
Hey there, Do I just remove the 757 line to get rid of the error?

**Reply by Cal-LF** [reply]
Don't remove it. Just comment it with // in front of the line.  //#define TMC_UART_PIN_E PA13

**Reply by smg492** [reply]
Hmm That gave me a failed error while trying to build.

**Reply by Bhumpe** [reply]
Did the error point out this line?
You placed 2 // direct in front and nothing else?

**Reply by smg492** [reply]
Yep, thats all I did 
![image](https://user-images.githubusercontent.com/84998374/120084497-af19c700-c09e-11eb-91d0-2ce1793c10e6.png)

**Reply by Bhumpe** [reply]
What does the build error say?

**Reply by smg492** [reply]
> What does the build error say?

![image](https://user-images.githubusercontent.com/84998374/120087863-ce721d80-c0b9-11eb-8b40-8d9ba64e48c9.png)

**Reply by Bhumpe** [reply]
One important thing to add, I did forget to mention this.
You need to add “_STANDALONE” on row 748!
See line 742 as reference.
That will fix this error!
![IMG_2016](https://user-images.githubusercontent.com/84877367/120096786-b4722280-c12d-11eb-967f-bce2356d9eec.jpeg)

**Reply by smg492** [reply]
That did it!  Thanks so much !!!
